pkg/rc/replication_controller.go

```diff
@@ -0,0 +1,40 @@
+package rc
+
+type ReplicationController interface {
+	// It is expected that the below functions will be used by the orchestrator,
+	// in response to user-requested deploy events.
+	// They affect the desired behavior of the replication controller.
+
+	// IncreaseReplicas(n) increases the number of desired replicas by n.
+	// This returns the new desired number if successful, or the old number and an error.
+	// It is expected that the replication controller will attempt to schedule its pod onto a node returned by EligibleNodes().
+	IncreaseReplicas(n int) (int, error)
+
+	// DecreaseReplicas(n) decreases the number of desired replicas by n.
+	// This returns the new desired number if successful, or the old number and an error.
+	DecreaseReplicas(n int) (int, error)
+
+	// Disable() causes the replication controller to stop reacting to changes in the number of desired replicas.
+	Disable() error
+
+	// It is expected that the below functions are used by the replication controller
+	// to determine whether observed state matches desired state.
+
+	// EligibleNodes() returns the nodes this replication controller could possibly run on.
+	EligibleNodes() ([]string, error)
```

Comment:
yes, the RC I have in mind internally has a LabelSelector and uses it when this function is called